src/faaslet/Faaslet.cpp

```diff
@@ -146,6 +149,9 @@ void Faaslet::postBind(const faabric::Message& msg, bool force)
 
         module->bindToFunction(msg);
     } else if (conf.wasmVm == "wavm") {
+#if (FAASM_SGX)
+        logger->warn("no SGX support for WAVM. continuing without SGX.");
+#endif
```

Comment:
We should avoid using preprocessor directives to perform logic like this wherever possible. We have a `msg.issgx()` property for checking whether the user has requested SGX execution which can be used here.